op-verifier/contracts/OPVerifier.sol

```diff
@@ -1,43 +1,177 @@
+    );
+    error DeprecatedProof(
+        OPWitnessProofType proofType,
+        uint256 disputeGameIndex
+    );
+
+    IOptimismPortal public immutable optimismPortal;
+    uint256 public immutable disputeGameMinAge;
 
-    IL2OutputOracle public opOracle;
     string[] _gatewayURLs;
 
-    constructor(string[] memory urls, address outputOracle) {
+    constructor(
+        string[] memory urls,
+        address optimismPortalAddress,
+        uint256 minimumAge
+    ) {
         _gatewayURLs = urls;
-        opOracle = IL2OutputOracle(outputOracle);
+        optimismPortal = IOptimismPortal(optimismPortalAddress);
+        disputeGameMinAge = minimumAge;
     }
 
-    function gatewayURLs() external view returns(string[] memory) {
+    function gatewayURLs() external view returns (string[] memory) {
         return _gatewayURLs;
     }
 
-    function getStorageValues(address target, bytes32[] memory commands, bytes[] memory constants, bytes memory proof) external view returns(bytes[] memory values) {
-        (OPWitnessData memory opData, StateProof memory stateProof) = abi.decode(proof, (OPWitnessData, StateProof));
-        Types.OutputProposal memory l2out = opOracle.getL2Output(opData.l2OutputIndex);
-        bytes32 expectedRoot = Hashing.hashOutputRootProof(opData.outputRootProof);
-        if(l2out.outputRoot != expectedRoot) {
-            revert OutputRootMismatch(opData.l2OutputIndex, expectedRoot, l2out.outputRoot);
+    function getL2OracleOutput(uint256 index) internal view returns (bytes32) {
+        IL2OutputOracle opOracle = optimismPortal.l2Oracle();
+
+        // If opOracle address is zero then revert with deprecated
+        if (address(opOracle) == address(0))
+            revert DeprecatedProof(OPWitnessProofType.L2OutputOracle, index);
+
+        // Get L2 Output Proposal at index
+        Types.OutputProposal memory l2out = opOracle.getL2Output(index);
+        return l2out.outputRoot;
+    }
+
+    function getDisputeGameOutput(
+        uint256 index
+    ) internal view returns (bytes32) {
+        IDisputeGameFactory disputeGameFactory = optimismPortal
+            .disputeGameFactory();
+
+        // If disputeGameFactory address is zero then revert with deprecated
+        if (address(disputeGameFactory) == address(0))
+            revert DeprecatedProof(OPWitnessProofType.DisputeGame, index);
+
+        // Get gameType used for L2 withdrawal
+        GameType respectedGameType = optimismPortal.respectedGameType();
+
+        // Get dispute game at index
+        (
+            GameType gameType,
+            Timestamp gameCreationTimeRaw,
+            IDisputeGame gameProxy
+        ) = disputeGameFactory.gameAtIndex(index);
+        Claim outputRoot = gameProxy.rootClaim();
+
+        // Unwrap gameCreationTime to uint64
+        uint64 gameCreationTime;
+        assembly {
+            gameCreationTime := gameCreationTimeRaw
+        }
+
+        // Wait for challenger to challenge the dispute game
```

Comment:
We also need a max age, to prevent gateways from returning old results.

Reply:
Max age is implemented in OPOutputLookup contract I have built at https://github.com/Opti-domains/dispute-game-lookup/blob/main/src/OPOutputLookup.sol. So do L2OutputOracle now using min and max age.